charts/cp-kafka/templates/_helpers.tpl

```diff
@@ -52,4 +52,17 @@ else use user-provided URL
 {{- $zookeeperConnectOverride := (index .Values "configurationOverrides" "zookeeper.connect") }}
 {{- default $zookeeperConnect $zookeeperConnectOverride }}
 {{- end -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+
+{{/*
+Form the Advertised Listeners. We will use the value of nodeport.firstListenerPort to create the
+external advertised listeners if configurationOverrides.advertised.listeners is not set.
+*/}}
+{{- define "cp-kafka.configuration.advertised.listeners" }}
```

Comment:
What if nodePort is disabled ?
We should skip EXTERNAL in such a situation.

Reply:
No. It's not working that way. Those addresses are broadcasted and I'm getting exceptions from clients that "cannot connect to broker id: x"

Reply:
I have have ran helm --debug --dry-run with the chart prior to this PR and can see that the following is always set regardless of values.yaml configuration

export KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://${POD_IP}:9092,EXTERNAL://${HOST_IP}:$((31090 + ${KAFKA_BROKER_ID})) && \

all this PR does is facilitate replacing 31090 with the firstListenerPort, so i don't see how this can be causing any unexpected behaviour that was not already occurring. I have tested this extensively with K8S on-prem deployments.. are you deploying to public cloud?

Also for some reason it only picks one of the ports to broadcast on "{{- $advertisedListenersOverride := first (pluck "advertised.listeners" .Values.configurationOverrides) }}" so 31090 was always selected.

Reply:
@galvo 
Yes. but it's not correct behaviour when nodePort is disabled and there is open bug for that.
https://github.com/confluentinc/cp-helm-charts/issues/145

Reply:
Ok i understand now, i will have a look at that